From ae332c8c2de6f5dc298f433b5013600a7272da0c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:03:54 -0400
Subject: [powerpc] proc: switch to fixed_size_llseek()

Message-id: <1410545655-205645-6-git-send-email-dzickus@redhat.com>
Patchwork-id: 94061
O-Subject: [RHEL7 PATCH 005/626] proc_powerpc: switch to fixed_size_llseek()
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit b33159b7d249afae386b91025ef4ea1d7793fc1b
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Sun Jun 23 12:35:26 2013 +0400

    proc_powerpc: switch to fixed_size_llseek()

    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/proc_powerpc.c b/arch/powerpc/kernel/proc_powerpc.c
index feb8580..c30612a 100644
--- a/arch/powerpc/kernel/proc_powerpc.c
+++ b/arch/powerpc/kernel/proc_powerpc.c
@@ -29,25 +29,9 @@
 
 #ifdef CONFIG_PPC64
 
-static loff_t page_map_seek( struct file *file, loff_t off, int whence)
+static loff_t page_map_seek(struct file *file, loff_t off, int whence)
 {
- loff_t new;
- switch(whence) {
- case 0:
-  new = off;
-  break;
- case 1:
-  new = file->f_pos + off;
-  break;
- case 2:
-  new = PAGE_SIZE + off;
-  break;
- default:
-  return -EINVAL;
- }
- if ( new < 0 || new > PAGE_SIZE )
-  return -EINVAL;
- return (file->f_pos = new);
+ return fixed_size_llseek(file, off, whence, PAGE_SIZE);
 }
 
 static ssize_t page_map_read( struct file *file, char __user *buf, size_t nbytes,
-- 
1.7.1